File: artifacts/tulipindicators/translate/src/ti_indicator_info.rs
use crate::*;


pub struct TiIndicatorInfo {
    pub name: Option<String>,
    pub full_name: Option<String>,
    pub start: TiIndicatorStartFunction,
    pub indicator: TiIndicatorFunction,
    pub indicator_ref: TiIndicatorFunction,
    pub type_: i32,
    pub inputs: i32,
    pub options: i32,
    pub outputs: i32,
    pub input_names: [Option<String>; 16],
    pub option_names: [Option<String>; 16],
    pub output_names: [Option<String>; 16],
    pub stream_new: TiIndicatorStreamNew,
    pub stream_run: TiIndicatorStreamRun,
    pub stream_free: TiIndicatorStreamFree,
}

File: artifacts/tulipindicators/decompose/src/ti_indicator_info.c
typedef struct ti_indicator_info
{
  const char *name;
  const char *full_name;
  ti_indicator_start_function start;
  ti_indicator_function indicator;
  ti_indicator_function indicator_ref;
  int type;
  int inputs;
  int options;
  int outputs;
  const char *input_names[16];
  const char *option_names[16];
  const char *output_names[16];
  ti_indicator_stream_new stream_new;
  ti_indicator_stream_run stream_run;
  ti_indicator_stream_free stream_free;
} ti_indicator_info
/*** DEPENDENCIES:
typedef int (*ti_indicator_start_function)(const double *options)
----------------------------
typedef int (*ti_indicator_stream_run)(ti_stream *stream, int size, const double * const *inputs, double * const *outputs)
----------------------------
typedef int (*ti_indicator_function)(int size, const double * const *inputs, const double *options, double * const *outputs)
----------------------------
typedef void (*ti_indicator_stream_free)(ti_stream *stream)
----------------------------
typedef int (*ti_indicator_stream_new)(const double *options, ti_stream **stream)
----------------------------
***/


